### DISCUSSION TITLE
How rename 'malt_parameters'

### DISCUSSION THREAD

**Discussion by warlockhb** [bnpr/Malt#473] [Development]
I want to manage some values ​​of shaders by adding global parameters in the panel

like this :
```
for node_tree in bpy.data.node_groups:
    node_tree.malt_parameters['A'] = 0.5
```

if i get this to work :
```
bpy.data.node_groups['Node_Alpha'].malt_parameters['A']
bpy.data.node_groups['Node_Beta'].malt_parameters['A']
```

parameters should be prepared like this.
but, parameter name is different for each Node Group

if i copy some node in nodegroup and paste to another nodegroup, name of some node pasted is different
to solve this, I tried to edit directly in the bpy data, but this only resulted in shader node error.

please tell me the solution.
So, this is the reason why I asked about recompilation in the previous article.

A_parameter = panel_parameter(different for each scene)

**Comment by pragma37** [comment]
You can get the internal `malt_parameter` name from its socket :
`bpy.data.node_groups["Tree Name"].nodes["Node Name"].inputs["Socket Name"].get_source_global_reference()`.

You're not supposed to change the internal name and doing so will likely break things.
